lib/client-api/src/story_store.ts

```diff
@@ -4,25 +4,35 @@ import memoize from 'memoizerific';
 import debounce from 'lodash/debounce';
 import { stripIndents } from 'common-tags';
 
+import { Channel } from '@storybook/channels';
 import Events from '@storybook/core-events';
 import { logger } from '@storybook/client-logger';
+import { StoryContext, StoryFn, Parameters } from '@storybook/addons';
+import {
+  DecoratorFunction,
+  LegacyData,
+  LegacyItem,
+  StoreData,
+  AddStoryArgs,
+  StoreItem,
+  ErrorLike,
+} from './types';
 
 // TODO: these are copies from components/nav/lib
 // refactor to DRY
-
-const toKey = input =>
+const toKey = (input: string) =>
```

Comment:
Maybe typing can be stricter to avoid many `as string` in this file,
Something like `(input: string): string`? (but I'm sure there is a reason that prevented you to do that)

Reply:
Yeah, why isn't this typed as string per @gaetanmaisse 's comment?